lib/ecto/adapters/postgres/connection.ex

```diff
@@ -414,6 +414,15 @@ if Code.ensure_loaded?(Postgrex) do
     defp operator_to_boolean(:and), do: " AND "
     defp operator_to_boolean(:or), do: " OR "
 
+    # Don't wrap json functions in parentheses
+    defp paren_expr({:fragment, _, [{:raw, raw_expr} | _]} = expr, sources, query) do
+      if String.match?(raw_expr, ~r/^\s*jsonb?/i) do
```

Comment:
We are still checking for json. We should never ever check for something as specific as json. We should just change the call to `paren_expr` to `expr` in `get_source`.

Reply:
This is because I have not pushed anything new since yesterday yet. I am hoping to rework the PR later this week. I am fairly new to Ecto and still getting my head around how it is structured, apologies!

Reply:
Oh, sorry. I saw a commit and thought you updated it but it was just about keeping up ebert. :)